x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/enrichments/enrichment_by_type/asset_criticality.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { cloneDeep } from 'lodash';
+import {
+  ALERT_HOST_CRITICALITY,
+  ALERT_USER_CRITICALITY,
+} from '../../../../../../../common/field_maps/field_names';
+import { createSingleFieldMatchEnrichment } from '../create_single_field_match_enrichment';
+import type {
+  CreateCriticalityEnrichment,
+  DoesAssetCriticalityAvailable,
+  CreateEnrichmentFunction,
+} from '../types';
+import { getFieldValue } from '../utils/events';
+import { getAssetCriticalityIndex } from '../../../../../../../common/asset_criticality';
+
+export const doesAssetCriticalityIndexExist: DoesAssetCriticalityAvailable = async ({
```

Comment:
Is it possible to use the asset criticality data client for this?

https://github.com/elastic/kibana/blob/18f8a959b7ea4b6a4f50a76ac337bffd559402ea/x-pack/plugins/security_solution/server/lib/entity_analytics/asset_criticality/asset_criticality_data_client.ts#L50

Reply:
Yes, I think it can be a good idea to use an asset criticality data client, but it currently has Logger as a dependency., and in rule execution, we have a different type - `IRuleExecutionLogForExecutors`.

To be able to use just Logger, we need to pass it to all rule types, and then they will have 2 loggers, which can lead to confusion

Reply:
Ah yes I see, I guess we could consider making logger an optional dependency for the data client, but that would probably be out of the scope of this PR